### ISSUE TITLE
Codehighlighting

### ISSUE THREAD

**Issue by NelDav** [tfachmann/MdCollab#12]
Implement codehighlighting

**Comment by NelDav** [comment]
I am a bit confused. Does the whole detection of keywords need to be proccesed by the website?
I hoped that pulldown-cmark marks the keyword for us.

**Comment by tfachmann** [comment]
Code-highlighting is non-trivial.
The only job `pulldown-cmark` has to do, is fence it in an block.

The highlighting itself can be done by external scripts such as [highlight.js](https://highlightjs.org/).
We will very likely integrate highlight.js in the future.

One question to evaluate is whether it is feasible in real-time (usually a post-function to highlight code) or if it causes any trouble.

**Comment by NelDav** [comment]
I took a look to the markdown compiler which I am using in VS Code to create html sides.

This compiler wraps each single word of the code into a html element with a class which specifies the function of a word:

```html
<pre class="hljs">
	<code>
		<div>
			<span class="hljs-function">
				<span class="hljs-keyword">def</span><span class="hljs-title">gcd</span><span class="hljs-params">(a, b)</span>:</span>
    <span class="hljs-keyword">if</span> a == <span class="hljs-number">0</span>:
        <span class="hljs-keyword">return</span> b

    <span class="hljs-keyword">while</span> b != <span class="hljs-number">0</span>:
        r = a % b
        a = b
        b = r

    <span class="hljs-keyword">return</span> a
</div></code></pre>
```

I am wondering if this is also possible with `pulldown-cmark`.

**Comment by tfachmann** [comment]
Yes, this is a different method. Something similar is used by Rmarkdown.

It is not possible with `pulldown-cmark` but would need some additional research as it might be more applicable in a live-preview.
